### DISCUSSION TITLE
XIAO ESP32C6 with PlatformIO | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1689] [Q&A]
# XIAO ESP32C6 with PlatformIO | Seeed Studio Wiki

Seeed Studio XIAO ESP32C6 with PlatformIO

https://wiki.seeedstudio.com/xiao_esp32c6_with_platform_io/

**Comment by WebDesignerlol** [comment]
is this still acurate? my platformio.ini generates this stock when i select the board:
[env:seeed_xiao_esp32c6]
platform = espressif32
board = seeed_xiao_esp32c6
framework = espidf

also any way you guys can write a guide for integrating the esp32c6 with esphome? it seems to have issues. thanks!

**Comment by hlima** [comment]
getting this eror when building a sample program:
PackageException: Got the unrecognized status code '404' when downloaded https://github.com/espressif/arduinoesp32/releases/download/3.0.2/esp32arduinolibs3.0.2.zip

**Comment by ZeSly** [comment]
platform.ini  example is outdated. I'm using this :

```
[env:seeed_xiao_esp32c6]
platform = https://github.com/pioarduino/platform-espressif32/releases/download/stable/platform-espressif32.zip
framework = arduino
board = seeed_xiao_esp32c6
```

**Reply by nfs0619** [reply]
Thank you very much for your feedback!

We've tested the configuration you provided and confirmed that it works correctly. We will update the relevant content in our wiki accordingly after further verification.

Really appreciate your contribution to improving the documentation!

**Comment by tomasz90** [comment]
I get 'ZigbeeCore.h' file not found in main.cpp..
I am using platformio setup, made according to get started... How to fix it?

**Reply by HM-sur** [reply]
Could you please take a look at the contents of your platformio.ini file

**Comment by inb4ohnoes** [comment]
It seems the configuration here isn't quite correct. I dug through the provided repo and found the right one:

```[env:seeed_xiao_esp32_c6]
platform = https://github.com/Seeed-Studio/platform-seeedboards.git
board = seeed-xiao-esp32-c6
framework = arduino`

**Reply by nfs0619** [reply]
Thank you for your feedback! We will verify it and update the documentation accordingly.

**Reply by tomasz90** [reply]
@inb4ohnoes it doesnt work...
<img width="812" alt="Screenshot 2025-04-28 at 15 24 40" src="https://github.com/user-attachments/assets/9b7088da-b64f-4fa2-a64f-8d68f97c4539" />

**Reply by inb4ohnoes** [reply]
No, it works, you have a totally different problem. Make sure you have Zigbee.h included in your compile sources. Sometimes this can be fixed simply by restarting vscode.

**Reply by tomasz90** [reply]
I dont use vscode. Platformio should download those files when syncing or building project but it doesn't do it.

**Reply by inb4ohnoes** [reply]
In that case I'd recommend posting your full setup in the help forums (IDE, frameworks, libraries, build env, everything else) and maybe somebody can help you out. Good luck!

**Comment by kevinhuihui** [comment]
Hi, i tried so many times and error is shown as below, does anyone know the problem?

my platform.ini:

[env:seeed_xiao_esp32c6]
platform = https://github.com/Seeed-Studio/platform-seeedboards.git
board = seeed-xiao-esp32-c6
framework = arduino`



Resolving seeed_xiao_esp32c6 dependencies...
Platform Manager: Installing git+https://github.com/Seeed-Studio/platform-seeedboards.git
git version 2.49.0.windows.1
Cloning into 'C:\Users\kevin\.platformio\.cache\tmp\pkg-installing-o6ih5hew'...

Platform Manager: Seeed Studio@1.0.0+sha.48aa906 has been installed!
MissingPackageManifestError: Could not find one of 'package.json' manifest files in the package

esp configure_default_packages
True
True
True
pop toolchain-xtensa-esp-elf

**Reply by kevinhuihui** [reply]
any feedback from seeed studio?

**Reply by nfs0619** [reply]
Thank you for your feedback. We have received your message and are currently arranging for debugging. If you do not receive further updates, please feel free to reach out to our technical support at [techsupport@seeed.io](mailto:techsupport@seeed.io) for further assistance. Thank you for your support and understanding.

**Comment by ndastur** [comment]
The platformio settings do not work. The board is listed as not working with the ardinio framework. only espidf

**Reply by nfs0619** [reply]
Hi! Please create an Arduino project with another ESP32 board first (e.g., **ESP32C3** or **ESP32S3**), then modify the **platformio.ini** to the XIAO ESP32C6 settings. This will allow Arduino compilation to work.

<img width="1081" height="454" alt="企业微信截图_17634302618023" src="https://github.com/user-attachments/assets/32f0d255-1bc5-4329-b1d0-9c50199ad013" />

If you still have issues, feel free to contact us at **[techsupport@seeed.io](mailto:techsupport@seeed.io)**.

**Comment by nick2090afk** [comment]
When iam using the following code on platformio.ini file and saving it:

[env:seeed_xiao_esp32_c6]
platform = https://github.com/Seeed-Studio/platform-seeedboards.git
board = seeed-xiao-esp32-c6
framework = arduino

Iam getting the following error:
'package-postinstall.py' is not recognized as an internal or external command, operable program or batch file.

Is there any workaround?

**Reply by ArmyGuy255A** [reply]
I ran into a few issues with this and it turned out that a bunch of my files were being installed into OneDrive and they were all offloaded from my machine. Here's what I did to fix this problem:

platformio.ini file:
[env:seeed_xiao_esp32c6]
platform = https://github.com/pioarduino/platform-espressif32.git#develop
board = seeed_xiao_esp32c6
framework = arduino

1. Update my default project location for pio to a place that's not in OneDrive 
ref: (https://gist.github.com/shnam7/953d19da155ecb32b4a2923589d6748d)
   > pio settings get projects_dir
   > pio settings set projects_dir "New Directory"

2. Nuke broken packages

# 1) Nuke the broken package + any partially-installed toolchain
Remove-Item -Recurse -Force "$env:USERPROFILE\.platformio\packages\toolchain-riscv32-esp" -ErrorAction SilentlyContinue
Remove-Item -Recurse -Force "$env:USERPROFILE\.platformio\packages\.tmp" -ErrorAction SilentlyContinue

# 2) Nuke the cached download/install temp (common corruption source)
Remove-Item -Recurse -Force "$env:USERPROFILE\.platformio\.cache\tmp" -ErrorAction SilentlyContinue

# 3) Optional but often fixes “bad cached archive”
Remove-Item -Recurse -Force "$env:USERPROFILE\.platformio\.cache\downloads" -ErrorAction SilentlyContinue

3. Reinstall / rebuild PIO boards/toolchain
pio run -t clean
pio run -v

4. Reload VSCode 

5. Build and Upload should now work.

**Comment by hpsaturn** [comment]
Seems that this board is not full supported yet with PlatformIO. I reached to build a helloworld with wifi but not a full project. The hello world compile good with:

```
[env:xiao-esp32-c6]
platform = espressif32
; platform = espressif32 @ 7.0.1
board = seeed_xiao_esp32c6
framework = arduino
```

but if I uncomment the line with target 7.0.1, it fails and it can't reached the board name.
if you use the package commented in this documentation fails. 

With my personal project that use FOTA and other libraries, you have many weird errors like this:

```
.pio/libdeps/XIAO_C6/esp32FOTA/src/esp32FOTA.cpp:37:10: fatal error: mbedtls/md_internal.h: No such file or directory
37 | #include "mbedtls/md_internal.h"
```
I have a branch (pull request) where you can replicate the issue:
https://github.com/kike-canaries/canairio_firmware/pull/338

**Reply by FaiyuetCik** [reply]
Thank you for your feedback. Regarding the adaptation, we will ask the R&D team to double-check if these issues exist, but I don't think this is the reason why your project fails to run. Here are my suggestions for you:
Using pioarduino:

  ini
  [env:XIAO_C6]
  platform = https://github.com/pioarduino/platform-espressif32/releases/download/stable/platform-espressif32.zip
  board = seeed_xiao_esp32c6
  framework = arduino, espidf
  monitor_speed = 115200

  Meanwhile, update the FOTA dependency to the version that includes compatibility fixes for Arduino-ESP32 3. x:

  lib_deps =
   https://github.com/chrisjoyce911/esp32FOTA.git#00c16436ec54f8634d03ad235afb188bd4122e2d

  The ESP32-C6 has only one high-performance application core, so in this project you cannot pin FreeRTOS tasks to Core 1. Please use ARDUINO_RUNNING_CORE, or Do not fix the core of the task.

 I tested this configuration locally and was able to build, flash, and start CanAirIO successfully. The CLI info command also worked without causing a restart. However, depending on your local dependency cache and project configuration, additional updates to the project’s transitive dependencies may still be required. 

Please note that CanAirIO is a third-party project with its own dependencies, so further project-specific adaptation may still be required.  
Please continue the discussion on the Seeed Studio Forum : https://forum.seeedstudio.com/, you can go ahead and ask questions.

**Comment by hpsaturn** [comment]
I closed by mistake the previous pull request that sho the problem. But I did a new issue with it:
https://github.com/kike-canaries/canairio_firmware/issues/340
